### PR TITLE
fix: Err returned in Rust not properly passed to Dart

### LINKED ISSUE #1387: [Bug] Panic within Rust is not printed in the Dart exception.
### Describe the bug

Panic within Rust is not printed in the Dart exception. This was sent during the previous version 1.70. However, now `toString` only prints `Instance of 'FrbAnyhowException'`

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
-- BEFORE

FfiException(RESULT_ERROR, previous `if` had no `then`
       --> <anon>:7:5
        |
      7 |     (i32.add
        |     ^, null)
  package:flutter_rust_bridge/src/basic.dart 129:9  FlutterRustBridgeBase._transformRust2DartMessage
  package:flutter_rust_bridge/src/basic.dart 70:9   FlutterRustBridgeBase.executeNormal.<fn>
  ===== asynchronous gap ===========================
  package:wasm_run_example/main.dart 32:14          getBinary
  ===== asynchronous gap ===========================
  package:wasm_run_example/main.dart 644:21         testAll.<fn>


-- AFTER

 test/main_test.dart: fueling instance execution limit (failed)
  Instance of 'FrbAnyhowException'
  package:flutter_rust_bridge/src/basic.dart 133:9  FlutterRustBridgeBase._transformRust2DartMessage
  package:flutter_rust_bridge/src/basic.dart 70:51  FlutterRustBridgeBase.executeNormal.<fn>
  ===== asynchronous gap ===========================
  package:wasm_run_example/main.dart 39:14          getBinary
  ===== asynchronous gap ===========================
  package:wasm_run_example/main.dart 790:21         testAll.<fn>
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

### LINKED ISSUE #1389: [Bug] Err returned in Rust not properly passed to Dart
### Describe the bug

After upgrading to version 1.82.1, the Err returned by a function signature containing a StreamSink was not correctly passed to Dart.

For example, the following Rust code:
```rust
pub fn require_run_lock(_sink: StreamSink<()>) -> Result<()> {
     return Err(anyhow::Error::msg("already require run lock"));
}
```

In 1.81.0, the exception information thrown by Dart is `FfiException(RESULT_ERROR, already require run lock, null)`

In 1.82.1, the exception message thrown by Dart is `Exception: tried to parse data but function is null`

Here is a minimal example to reproduce the issue: https://github.com/OfficialBoyfriend/rust_demo, just switch to 1.82.1 to revert the issue.

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
-
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by OfficialBoyfriend**
## Changes

Fixes #1387
Fixes #1389 

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (via `just precommit`).
- [ ] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

## Remark for PR creator

- Justfile is a task runner for the command line interface that allows you to run shell commands from a file named `justfile` in your project directory. You can use Justfile after [installing it](https://github.com/casey/just). Note that commands written in `justfile` of this repository are expected to be run in `bash`, not `cmd` or `powershell`. Running `just ...` commands in `cmd` or `powershell` will produce errors as the syntax is not compatible. On Windows, you can use `git bash` if you have Git installed.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job! I will review after the checklist shown above are finished, more specifically:

- [ ] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [ ] The code generator is run and the code is formatted (via `just precommit`).
- [ ] CI is passing.

**Comment by OfficialBoyfriend** [comment]
> Good job! I will review after the checklist shown above are finished, more specifically:
> 
>     * [ ]  New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
> 
>     * [ ]  The code generator is run and the code is formatted (via `just precommit`).
> 
>     * [ ]  CI is passing.

@fzyzcjy  Check, please. The two failed tasks in CI appear to have nothing to do with the changes I made.

**Comment by fzyzcjy** [comment]
> The two failed tasks in CI appear to have nothing to do with the changes I made.

That's true. Reviewing!

File: frb_example/pure_dart/dart/lib/main.dart [hunk 1]
```diff
@@ -1502,6 +1502,19 @@ void main(List<String> args) async {
...
+      try {
+        await for (final _ in api.streamSinkThrowAnyhow()) {
+          break;
+        }
+        assert(false);
+      } catch (e) {
+        final FrbAnyhowException p = e as FrbAnyhowException;
+        print("anyhow error: ${p.anyhow}");
+        assert(p.anyhow.contains("anyhow error"));
+      }
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/lib/main.dart:1516]: Maybe sth like

```
expect(
  () async { await for (final _ in api.streamSinkThrowAnyhow()) {} },
  throwsA(isA<FrbAnyhowException>())
    .having('toString', (e) => e.toString(), 'anyhow error')
);
```

File: frb_dart/lib/src/basic.dart [hunk 2]
```diff
@@ -234,6 +234,9 @@ class FrbAnyhowException implements FrbException {
   final String anyhow;
 
   FrbAnyhowException(this.anyhow);
+
+  @override
+  String toString() => anyhow;
```
Comment (fzyzcjy) [inline frb_dart/lib/src/basic.dart:239]: Maybe the following toString is a bit more aligned with flutter (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#override-tostring):

```suggestion
  String toString() => 'FrbAnyhowException($anyhow)';
```

File: frb_codegen/src/parser/mod.rs [hunk 3]
```diff
@@ -294,35 +293,41 @@ impl<'a> Parser<'a> {
...
+        }
+
+        if output.is_none() {
+            let tmp_mode = if let IrType::SyncReturn(_) = result.0 {
                 IrFuncMode::Sync
             } else {
                 IrFuncMode::Normal
-            });
+            };
+            mode = Some(tmp_mode);
```
Comment (fzyzcjy) [inline frb_codegen/src/parser/mod.rs:329]: nit: Looks like

```
mode = Some(if let ... { ... } else { ... });
```

and we do not need `tmp_mode`

**Review by fzyzcjy — APPROVED** [review]
Nice fix! I am ready to merge after nits

**Comment by fzyzcjy** [comment]
@all-contributors please add @OfficialBoyfriend for code

**Pull request merged**